TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -5,28 +5,73 @@
 using TutorLizard.BusinessLogic.Models.DTOs;
 using TutorLizard.BusinessLogic.Models.DTOs.Requests;
 using TutorLizard.BusinessLogic.Models.DTOs.Responses;
+using TutorLizard.BusinessLogic.Interfaces.Repositories;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using Microsoft.EntityFrameworkCore;
+using System.Net.NetworkInformation;
+using System;
 
 namespace TutorLizard.BusinessLogic.Services;
 public class TutorService : ITutorService
 {
-    private readonly IDbRepository<ScheduleItem> _scheduleItemRepository;
     private readonly IDbRepository<ScheduleItemRequest> _scheduleItemRequestRepository;
-
+    private readonly IDbRepository<ScheduleItem> _scheduleItemRepository;
+    private readonly IDbRepository<Ad> _adRepository;
     public TutorService(IDbRepository<ScheduleItem> scheduleItemRepository,
-                        IDbRepository<ScheduleItemRequest> scheduleItemRequestRepository)
+                        IDbRepository<ScheduleItemRequest> scheduleItemRequestRepository,
+                        IDbRepository<Ad> adRepository)
     {
         _scheduleItemRepository = scheduleItemRepository;
         _scheduleItemRequestRepository = scheduleItemRequestRepository;
+        _adRepository = adRepository;
+    }
+
+    public async Task<IsUserTheAdOwnerResponse> IsUserTheAdOwner(IsUserTheAdOwnerRequest request)
+    {
+        int adId = request.AdId;
+        int userId = (int)request.UserId;
```

Comment:
Before you cast to int, check if it's not null and return early if it is. It can cause an exception.